linux/ibus-keyman/src/bcp47util.c

```diff
@@ -0,0 +1,31 @@
+#include <gio/gio.h>
+#include <unicode/locid.h>
+
+int bcp47_minimize(const char* tag, char* minimizedTag, int tagCapacity) {
+  UErrorCode status = U_ZERO_ERROR;
+  if (!tag || strlen(tag) == 0) {
+    strncpy(minimizedTag, "", tagCapacity);
+    return -1;
+  }
+  int capacity = 255;
+  char localeId[capacity];
+  uloc_forLanguageTag(tag, localeId, capacity, NULL, &status);
+  if (U_FAILURE(status)) {
+    g_error("%s: uloc_forLanguageTag returned %0x", __FUNCTION__, status);
+    return -1;
+  }
+
+  char minimizedLocaleId[capacity];
+  uloc_minimizeSubtags(localeId, minimizedLocaleId, capacity, &status);
```

Comment:
Does Keyman Developer use langtags.json to determine canonical tags? Wondering if we use that instead of icu? (would require more code though

Reply:
We are moving towards using ICU for canonicalization everywhere. But we have work to do.

Reply:
what was it returning with `und` as input?